maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/PluginsMetadataGenerator.java

```diff
@@ -124,7 +126,50 @@ public Collection<? extends Metadata> finish( Collection<? extends Artifact> art
                 }
             }
         }
-
         return plugins.values();
     }
+
+    private PluginInfo extractPluginInfo( Artifact artifact )
+    {
+        // sanity: jar, no classifier and file exists
+        if ( artifact != null
+                && "jar".equals( artifact.getExtension() )
+                && "".equals( artifact.getClassifier() )
+                && artifact.getFile() != null )
+        {
+            Path artifactPath = artifact.getFile().toPath();
+            if ( Files.isRegularFile( artifactPath ) )
```

Comment:
What if the plugin is the reactor and the jar hasn't been built? Can this happen?

Reply:
PluginsMetadataGenerator <-isInvokedBy- PluginsMetadataGeneratorFactory <-isInvokedBy- Deployer... hence, your question is "what if a deploy is ongoing for a project that is not built" that is a contradiction (or just bad API usage of resolver?)

Reply:
Thanks, that is what I wanted to hear. We won't get so far w/o building the JAR.